webapp/src/js/controllers/reports-content.js

```diff
@@ -108,15 +114,15 @@ var _ = require('underscore');
               $scope.deselectReport(change.doc);
             });
           } else {
-            var selected = $scope.selected;
+            var selected = ctrl.selected;
             $scope.refreshReportSilently(change.doc)
               .then(function() {
                 if(selected[0].formatted.verified !== change.doc.verified ||
                    ('oldVerified' in selected[0].formatted &&
                     selected[0].formatted.oldVerified !== change.doc.verified)) {
-                  $scope.selected = selected;
+                  ctrl.setSelected(selected);
                   $timeout(function() {
-                    $scope.selected[0].formatted.verified = change.doc.verified;
+                    ctrl.setFirstSelectedFormattedProperty({ verified: change.doc.verified });
```

Comment:
`setFirstSelectedFormattedProperty` here and `setFirstSelectedDocProperty` below are pretty hideous names and their corresponding code in `reducers/index.js` are equally not nice. Unfortunately `lodash.merge` (which is like a deep version of `Object.assign`) which is used for `updateSelected` won't override values with `undefined`, which most of the time is fine, but here we occasionally set `verified` back to `undefined`, and so `lodash.merge` can't be used. Using `Object.assign` then is the only way I can see and results in some specific and ugly code.